presto-main/src/main/java/com/facebook/presto/util/MergeSortedPages.java

```diff
@@ -65,6 +71,14 @@ private MergeSortedPages() {}
             AggregatedMemoryContext aggregatedMemoryContext,
             DriverYieldSignal yieldSignal)
     {
+        requireNonNull(pageProducers, "pageProducers is null");
```

Comment:
doesn't hurt but is not necessary. Generally, the practise is to `requireNonNull` when otherwise NPE would not occur until some time later (e.g. in constructors, around lambda captures)

Reply:
> Generally, the practise is to requireNonNull when otherwise NPE would not occur until some time later (e.g. in constructors, around lambda captures)

This is true for `comparator`, `outputChannels`, `pageBreakPredicate` and `yieldSignal`. I think it makes sense to check them all then

Reply:
`comparator` -- yes.

`outputChannels`, `pageBreakPredicate`, `yieldSignal` -- these are passed through